src/Cake.Core/IO/DirectoryPath.cs

```diff
@@ -51,6 +51,24 @@ public FilePath GetFilePath(FilePath path)
             return new FilePath(PathHelper.Combine(FullPath, path.GetFilename().FullPath));
         }
 
+        /// <summary>
+        /// Gets the directory path of a <see cref="DirectoryPath"/>.
+        /// </summary>
+        /// <returns>A <see cref="DirectoryPath"/> to the parent directory of the given <see cref="DirectoryPath"/>.</returns>
+        public DirectoryPath GetParent()
+        {
+            var segments = Segments.Take(Segments.Length - 1).ToArray();
+            var path = PathHelper.Combine(segments);
+
+            // If it was an absolute path with no parent.
+            if (path == @"\\")
+            {
+                return null;
+            }
```

Comment:
Couldn't we check the segment length instead before combining the segments? If there are no parents, `segments` will be `0`, right?

Reply:
I just went through with some breakpoints, and this turns out to be false. Segments length will be 1 or 2 if there is no parent, depending on if the input begins with '\\\\'. As the code currently stands, it handles these cases appropriately, though there may be a better way :)

Reply:
ah, `segments` would be `0`, unless it is `@"\\"` in which case it would be `1`. This could also be a valid parent, so I think it's best to check for this particular case from what I understand.